cmake/machinecfg/RZAnsel.cmake

```
@@ -0,0 +1,214 @@
+#========================================================================================
+# Parthenon performance portable AMR framework
+# Copyright(C) 2020 The Parthenon collaboration
+# Licensed under the 3-clause BSD License, see LICENSE file for details
+#========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#========================================================================================
+
+# OPTIONS:
+# - `RZANSEL_VIEW_DATE` - The date the dependencies were installed.
+#       Default:
+#           unset, which results in using the view associated with your
+#           current commit. See `RZANSEL_VIEW_DATE_LATEST`
+# - `RZANSEL_COMPILER` - Compiler family to use
+#       Default: "GCC"
+#       Possible Values: "GCC", "GCC8"
+# - `RZANSEL_CUDA` - Build for CUDA
+#       Default: ON 
+# - `RZANSEL_PROJECT_PREFIX`
+#   Description: [ADVANCED] Point to an alternative parthenon-project path
+#       Default: /usr/gapps/parthenon_shared/parthenon-project
+
+cmake_minimum_required(VERSION 3.15)
+
+message(STATUS "Loading machine configuration for LNLL RZAnsel.\n"
+  "Supported MACHINE_VARIANT includes 'cuda', 'mpi', and 'cuda-mpi'\n")
+
+set(RZANSEL_ARCH "ppc64le")
+# NOTE: When updating dependencies with new compilers or packages, you should
+# ideally only need to update these variables to change the default behavior.
+set(RZANSEL_VIEW_DATE_LATEST "2021-01-04")
+set(RZANSEL_GCC_PREFERRED "GCC8")
+set(RZANSEL_COMPILER_PREFERRED "GCC")
+set(RZANSEL_GCC8_VERSION "8.3.1")
+set(RZANSEL_VARIANT_PREFIX "_no-mpi")
+set(RZANSEL_MPI_DATE "2020.08.19")
+set(MPIEXEC_EXECUTABLE "/usr/tcetmp/bin/jsrun" CACHE STRING "Mpi executable to use on RZansel system")
+set(RZANSEL_CUDA_VERSION "10.1.243")
+set(MACHINE_VARIANT "cuda-mpi" CACHE STRING "Machine variant to use when building on RZAnsel")
+set(NUM_RANKS 4) 
+set(GPU_COUNT "4")
+
+if (${MACHINE_VARIANT} MATCHES "cuda")
+  set(RZANSEL_CUDA_DEFAULT ON)
+else()
+  set(RZANSEL_CUDA_DEFAULT OFF)
+endif()
+
+set(RZANSEL_CUDA ${RZANSEL_CUDA_DEFAULT} CACHE BOOL "Build for CUDA")
```

```suggestion
if (${MACHINE_VARIANT} MATCHES "cuda")
  set(RZANSEL_CUDA ON CACHE BOOL "Build for CUDA")
else()
  set(RZANSEL_CUDA OFF CACHE BOOL "Build for CUDA")
endif()
```

I don't see `RZANSEL_CUDA_DEFAULT` being used anywhere else. So if I'm not missing anything one could skip that intermediate variable.